pool-payout-zkapp/src/PoolPayout.ts

```diff
@@ -161,25 +206,23 @@ export class PoolPayout extends SmartContract {
 
     // Debugging control flow
     // If we are at the number of delegators we can send the fees to the onchain validated public key
-    const checkBigger = Circuit.if(
-      currentIndex.gte(feePayout.numDelegates),
-      (() => {
-        // TRUE
-        this.currentIndex.set(Field(0));
-        this.currentEpoch.set(epoch.add(1));
-        this.send({
```

Comment:
Can't call `send` in an if block since both branches are actually executed.  Only the value of the true branch gets returned, but both sides get run.

Reply:
Thanks for the explanation, that makes a lot of sense.